src/models/event.js

```diff
@@ -3,14 +3,67 @@ import Attributes from './attributes';
 import EventParticipant from './event-participant';
 
 export default class Event extends RestfulModel {
+  
+  get start() {
+    return 
+      this.when.start_time ||
+      this.when.start_date ||
+      this.when.time ||
+      this.when.date;
+  }
+
+  get end() {
+    return
+      this.when.end_time ||
+      this.when.end_date ||
+      this.when.time ||
+      this.when.date;
+  }
+
+  set start(val) {
+    if (this.when) {
+      if (typeof val === 'number') {
+        if (val === this.when.end_time) {
+          this.when = { time: val }
+        } else {
+          delete this.when.time;
+          this.when.start_time = val;        
+        }   
+      }
+      if (typeof val === 'string') {
+        if (val === this.when.end_date) {
+          this.when = { date: val }
+        } else {
+          delete this.when.date;
+          this.when.start_date = val;
+        } 
+      }
+    } 
+  }
+
+  set end(val) {
+    if (this.when) {
+      if (typeof val === 'number') {
+        if (val === this.when.start_time) {
+          this.when = { time: val }
+        } else {
+          delete this.when.time;
+          this.when.end_time = val;        
+        }   
+      }
+      if (typeof val === 'string') {
+        if (val === this.when.start_date) {
+          this.when = { date: val }
+        } else {
+          delete this.when.date;
+          this.when.end_date = val;
+        } 
+      }
+    } 
+  }
+
   saveRequestBody() {
     const dct = this.toJSON();
-    if (this.start && this.end) {
-      dct['when'] = {
-        start_time: this.start.toString(),
-        end_time: this.end.toString(),
-      };
-    }
     delete dct['_start'];
     delete dct['_end'];
```

Comment:
let's rm these too, i dont think they'll be there anymore

Reply:
good point! i'm going to delete the entire saveRequestBody() method on the event class then since there's nothing unique that needs to be added and the event class inherits saveRequestBody() from the restful model.